js/src/admin/components/ExtensionPage.js

```diff
@@ -0,0 +1,251 @@
+import Alert from '../../common/components/Alert';
+import Page from '../../common/components/Page';
+import Button from '../../common/components/Button';
+import icon from '../../common/helpers/icon';
+import ItemList from '../../common/utils/ItemList';
+import listItems from '../../common/helpers/listItems';
+import LoadingModal from './LoadingModal';
+import ExtensionPermissionGrid from './ExtensionPermissionGrid';
+import addExtensionPermission from '../utils/addExtensionPermission';
+import Switch from '../../common/components/Switch';
+import saveSettings from '../utils/saveSettings';
+
+export default class ExtensionPage extends Page {
```

Comment:
How does this look when the extension is disabled? I suppose the form wouldn't show up then (since js isn't registered)?  Might be better to have a special message for that to justify the blank page.